theatre/core/src/propTypes/index.ts

```diff
@@ -234,6 +242,100 @@ const _interpolateNumber = (
   return left + progression * (right - left)
 }
 
+export const rgba = (
+  defaultValue: Rgba,
+  opts?: {
+    label?: string
+    sanitize?: Sanitizer<Rgba>
+    interpolate?: Interpolator<Rgba>
+  },
+): PropTypeConfig_Rgba => {
+  if (process.env.NODE_ENV !== 'production') {
+    validateCommonOpts('t.rgba(defaultValue, opts)', opts)
+
+    // Lots of duplicated code and stuff that probably shouldn't be here, mostly
+    // because we are still figuring out how we are doing validation, sanitization,
+    // decoding, decorating.
+
+    // Validate default value
+    let valid = true
+    for (const p of ['r', 'g', 'b', 'a']) {
+      if (
+        !Object.prototype.hasOwnProperty.call(defaultValue, p) ||
+        typeof (defaultValue as $IntentionalAny)[p] !== 'number'
+      ) {
+        valid = false
+      }
+    }
+
+    if (!valid) {
+      throw new Error(
+        `Argument defaultValue in t.rgba(defaultValue) must be of the shape { r: number; g: number, b: number, a: number; }.`,
+      )
+    }
+
+    // Clamp defaultValue components between 0 and 1
+    for (const component of ['r', 'g', 'b', 'a']) {
+      ;(defaultValue as $IntentionalAny)[component] = Math.min(
```

Comment:
Caller should expect args to be immutable. So let's either throw a validation error, or create a new object.

Reply:
Yeah, thought I'd cheat here for performance, good point